src/gameGlobalInfo.cpp

```diff
@@ -202,17 +202,30 @@ string getSectorName(sf::Vector2f position)
     constexpr float sector_size = 20000;
     int sector_x = floorf(position.x / sector_size) + 5;
     int sector_y = floorf(position.y / sector_size) + 5;
-    string y;
-    string x;
-    if (sector_y >= 0)
-        y = string(char('A' + (sector_y)));
-    else
-        y = string(char('z' + sector_y / 26)) + string(char('z' + 1 + (sector_y % 26)));
-    if (sector_x >= 0)
-        x = string(sector_x);
-    else
-        x = string(100 + sector_x);
-    return y + x;
+
+    int quadrant = 0;
+    string row = "";
+    if (sector_y < 0)
+    {
+        quadrant += 2;
+        sector_y = -1 - sector_y;
+    }
+    if (sector_x < 0)
+    {
+        quadrant += 1;
+        sector_x = -1 - sector_x;
+    }
+    while (sector_y > 0)
+    {
+        int digit = (sector_y - 1) % 26;
+        row = (char)('A' + digit) + row;
+        sector_y = (int)((sector_y - digit) / 26);
+    }
+
+    char column[(sector_x / 10) + 1];
+    sprintf(column, "%d", (int)sector_x);
```

Comment:
Perfect example of C code being bad for people. The buffer you allocate can be really big if the sector_x value is really large. Which is a waste.
There is also no need for sprintf, as the SeriousProton "string" class has a constructor that converts an integer into a string. (which was used by the original code)

Reply:
yes, I was wondering about that magic... will fix.

Reply:
regarding the size of the buffer, if the scroll is infinite, then either the sector names repeat themselves or they keep growing. I think growing is the way to go.

Reply:
oh, I  think i got what you mean about the buffer. it should be `std::log10(sector_x)`, right?